### PR TITLE
Bundle Source Link packages

### PR DESCRIPTION
**Pull request by tmat**
Source Link targets are included automatically unless the project has already a `PackageReference` to any Source Link package.

Property `SuppressImplicitGitSourceLink` can be used to disable automatic inclusion of Source Link.

### CODE REVIEW

File: src/Assets/TestProjects/SourceLinkTestApp/SourceLinkTestApp.csproj [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <RestorePackagesPath Condition=" '$(RestorePackagesPath)' == '' ">$(MSBuildThisFileDirectory)..\..\..\artifacts\.nuget\packages</RestorePackagesPath>
```
[the remark this answers is not in the document]
Comment (dsplaisted) [reply src/Assets/TestProjects/SourceLinkTestApp/SourceLinkTestApp.csproj:3]: Looking at the tests, it looks like this project is copied to the temp folder, so the relative path will be wrong anyway.

File: src/Tests/Microsoft.NET.Build.Tests/SourceLinkTests.cs [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute().Should().Pass();
+
+            var intermediateDir = buildCommand.GetIntermediateDirectory();
+            intermediateDir.Should().NotHaveFile("SourceLinkTestApp.sourcelink.json");
+        }
+
+        /// <summary>
+        /// When creating a new repository locally we want the build to work and not report warnings even before the remote is set.
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/SourceLinkTests.cs:89]: This description (that the build shouldn't report warnings) seems to contradict the assertion (that it should include a warning).

**Review by dsplaisted — APPROVED** [review]
I haven't looked at the code for each sourcelink package, so I'm assuming those are good to be integrated into the SDK.

Once this flows into dotnet/installer, we should check to see whether the sourcelink SDKs are appropriately crossgen'd.

File: src/Tests/Microsoft.NET.Build.Tests/SourceLinkTests.cs (hunk 2 above)
Comment (tmat) [reply src/Tests/Microsoft.NET.Build.Tests/SourceLinkTests.cs:89]: Ah, forgot to update the assert. Was testing it manually.

File: src/Layout/redist/targets/RestoreDependency.proj [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<Project ToolsVersion="15.0" DefaultTargets="EnsureDependencyRestored;CopySdkToOutput">
+﻿<Project ToolsVersion="15.0" DefaultTargets="PrepareBundledDependencyProps;EnsureDependencyRestored;CopySdkToOutput">
```
[the remark this answers is not in the document]
Comment (tmat) [reply src/Layout/redist/targets/RestoreDependency.proj:1]: Removed from dependencies

**Review by dsplaisted — APPROVED** [review]

**Pull request merged**